File: Sort Letters by Case.cpp
class Solution {
public:
    /** 
     * @param chars: The letters array you should sort.
     */
    void sortLetters(string &letters) {
        // write your code here
        int len = letters.length();
        int left = 0, right = len - 1;
        while (left < right)
        {
            while (left < right && islower(letters[left]))
            {
                ++left;
            }
            while (left < right && isupper(letters[right]))
            {
                --right;
            }
            swap(letters[left], letters[right]);
            ++left;
            --right;
        }
    }
};

File: Sort Letters by Case.py
class Solution:
    """
    @param chars: The letters array you should sort.
    """
    def sortLetters(self, chars):
        # write your code here
        if not chars:
            return None
        
        length = len(chars)
        left, pivot, right = 0, 0, length - 1
        while left <= right:
            if chars[left].islower():
                chars[pivot], chars[left] = chars[left], chars[pivot]
                pivot += 1
                left += 1
            else:
                chars[left], chars[right] = chars[right], chars[left]
                right -= 1
        
